client/src/actions/bulk.js

```diff
@@ -0,0 +1,69 @@
+import {
+    BULK_VALIDATE_MONUMENTS_PENDING, BULK_VALIDATE_MONUMENTS_SUCCESS, BULK_VALIDATE_MONUMENTS_ERROR,
+    BULK_CREATE_MONUMENTS_PENDING, BULK_CREATE_MONUMENTS_SUCCESS, BULK_CREATE_MONUMENTS_ERROR
+} from '../constants';
+import { post, postFile } from '../utils/api-util';
+import { addError } from './errors';
+
+const actions = {
+    validate: 'validate',
+    create: 'create'
+};
+
+function pending(action) {
+    return {
+        type: action === actions.validate ? BULK_VALIDATE_MONUMENTS_PENDING : BULK_CREATE_MONUMENTS_PENDING
+    };
+}
+
+function success(action, result) {
+    return {
+        type: action === actions.validate ? BULK_VALIDATE_MONUMENTS_SUCCESS : BULK_CREATE_MONUMENTS_SUCCESS,
+        payload: result
+    };
+}
+
+function error(action, error) {
+    return {
+        type: action === actions.validate ? BULK_VALIDATE_MONUMENTS_ERROR : BULK_CREATE_MONUMENTS_ERROR,
+        error: error
+    };
+}
```

Comment:
I like this format when we have multiple action types but with the same basic functionality.

Reply:
I just thought of an even better way of doing this:
```
const actions = {
    validate: {
        pending: BULK_VALIDATE_MONUMENTS_PENDING,
        success: BULK_VALIDATE_MONUMENTS_SUCCESS,
        error: BULK_VALIDATE_MONUMENTS_ERROR
    },
    create: // etc.
};

...

function error(action, error) {
    return {
        type: action.error,
        error: error
    }
}
...
error(actions.validate, error);
```